packages/neuron-wallet/src/services/transactions.ts

```diff
@@ -89,20 +114,33 @@ export default class TransactionsService {
   public static getAllByAddresses = async (
     params: TransactionsByAddressesParam,
   ): Promise<PaginationResult<Transaction>> => {
+    const lockHashes: string[] = []
+    for (const addr of params.addresses) {
+      const lockHash: string = await TransactionsService.addressToLockHash(addr)
+      lockHashes.push(lockHash)
+    }
+
     return TransactionsService.getAll({
       pageNo: params.pageNo,
       pageSize: params.pageSize,
-      lockHashes: [],
+      lockHashes,
     })
   }
 
   public static getAllByPubkeys = async (
     params: TransactionsByPubkeysParams,
   ): Promise<PaginationResult<Transaction>> => {
+    const lockHashes: string[] = []
```

Comment:
Does `map` work for this loop (with `await` inside)?

Reply:
Yes, I'll update it.

Reply:
Updated